osu.Game/Rulesets/Scoring/Score.cs

```diff
@@ -28,9 +29,21 @@ public class Score
         public int Combo { get; set; }
 
         [JsonProperty(@"mods")]
-        protected string[] ModStrings { get; set; } //todo: parse to Mod objects
+        private string[] modStrings { get; set; }
 
-        public RulesetInfo Ruleset { get; set; }
+        private RulesetInfo ruleset;
+        public RulesetInfo Ruleset
+        {
+            get { return ruleset; }
+            set
+            {
+                ruleset = value;
+
+                // Handle the mod strings if they are assigned
+                if (modStrings != null)
+                    Mods = ruleset.CreateInstance().GetAllMods().Where(mod => modStrings.Contains(mod.ShortenedName)).ToArray();

```

Comment:
`Mods` should instead be made into a property, because it's going to need to set the mods array also anyway.